NODE-916: Run local LRT with Docker

igor-ramazanov opened the pull request:
### Overview
Adds a new `lrt.sh` Bash script to run a local long-running test using the `hack/docker` setup. It may be useful for a faster debugging process.

### Which JIRA ticket does this PR relate to?
https://casperlabs.atlassian.net/browse/NODE-916

### Complete this checklist before you submit this PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [CasperLabs coding standards](https://casperlabs.atlassian.net/wiki/spaces/EN/pages/16842753/Coding+Standards).
- [ ] If this PR adds a new feature, it includes tests related to this feature.
- [x] You assigned one person to review this PR.
- [x] Your GitHub account is linked with our [Drone CI](https://drone-auto.casperlabs.io/) system. This is necessary to run tests on this PR.

### Notes
_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._

aakoshh commented:
I think this covers https://casperlabs.atlassian.net/browse/NODE-916

aakoshh commented on hack/docker/Makefile:
```diff
@@ -106,6 +106,7 @@ slow:
 	../key-management/docker-gen-account-keys.sh .casperlabs/genesis/system-account
 	../key-management/docker-gen-account-keys.sh keys/account-0
 	../key-management/docker-gen-keys.sh .casperlabs/node-0
+	cp -r .casperlabs/genesis/system-account keys/system-account
```
FYI in my PR https://github.com/CasperLabs/CasperLabs/pull/1142 this account no longer exists.

igor-ramazanov replied:
Hm, I think the best way to properly handle these changes is to merge this PR into the `dev` (if it goes before #1142) and then merge `dev` into #1142 and make it work there.

It could be more comfortable if #1142 got merged first.

aakoshh replied:
Yeah, whichever goes first, easy fix either way.

aakoshh commented on hack/docker/lrt.sh:
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+# TODO: Use Makefile instead? Something like `make lrt`.
```
I think it's fine this way, easier to pass parameters and the Makefile does enough already setting up all the pieces.

igor-ramazanov replied:
Will remove the TODO

igor-ramazanov replied:
Removed

aakoshh commented on hack/docker/template/.env:
```diff
@@ -1,3 +1,3 @@
 CL_LOG_LEVEL=INFO
 CL_VERSION=latest
-CL_CASPER_AUTO_PROPOSE_ENABLED=false
+CL_CASPER_AUTO_PROPOSE_ENABLED=true
```
This changes the default which is means all the examples we have that tell the user to use `propose` are now wrong as it would be no longer needed. I left it deliberately on `false` so people can keep control during demos and stuff. 

If all you want is for the feature to be on during the LRT you wrote, why don't you just start it in `lrt.sh` like this?
```
CL_CASPER_AUTO_PROPOSE_ENABLED=true make up node-0/up node-1/up node-2/up
```

igor-ramazanov replied:
Oh, thanks! I didn't know I can do that.

igor-ramazanov replied:
Done

aakoshh commented on hack/docker/template/start-node.sh:
```diff
@@ -5,7 +5,8 @@ set -e
     exec ./bin/casperlabs-node run -s \
-        --server-host $HOSTNAME
+        --server-host $HOSTNAME \
+        --casper-initial-motes=10000000000000000000000000
```
This option is gone in my PR as well. Also the template already sets this at https://github.com/CasperLabs/CasperLabs/blob/dev/hack/docker/template/docker-compose.yml#L48

Let's stick to using that place for variables, they are easier to see in one place, and leave the startup script for things that are really about bootstrap vs non-bootstrap nodes.

igor-ramazanov replied:
Done

igor-ramazanov commented:
@aakoshh I addressed the comments and also changed the commits messages reflecting the ticket name. Unfortunately, I can't change the branch name without closing the PR and creating a new one.

igor-ramazanov commented:
bors r+
